explorer/src/pages/inspector/RawInputCard.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import { Message } from "@solana/web3.js";
+import type { TransactionData } from "./InspectorPage";
+import { useQuery } from "utils/url";
+import { useHistory, useLocation } from "react-router";
+
+export function RawInput({
+  value,
+  setMessage,
+}: {
+  value?: string;
+  setMessage: (param: TransactionData | undefined) => void;
+}) {
+  const rawTransactionInput = React.useRef<HTMLTextAreaElement>(null);
+  const [error, setError] = React.useState();
+  const [rows, setRows] = React.useState(3);
+  const query = useQuery();
+  const history = useHistory();
+  const location = useLocation();
+
+  const onInput = React.useCallback(() => {
+    const base64 = rawTransactionInput.current?.value;
+    if (base64) {
+      if (query.get("message")) {
+        query.delete("message");
+        history.push({ ...location, search: query.toString() });
+      }
+
+      setRows(Math.max(3, Math.min(10, Math.round(base64.length / 150))));
+
+      let buffer;
+      try {
+        buffer = Uint8Array.from(atob(base64), (c) => c.charCodeAt(0));
+      } catch (err) {
+        console.error(err);
+        setError(err);
```

Comment:
nit: The exception thrown by atob isn't very user friendly: "Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded."

Reply:
Thanks, changed to `"Input must be base64 encoded"`